codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/service/ServiceStubGenerator.kt

```diff
@@ -0,0 +1,82 @@
+package software.amazon.smithy.kotlin.codegen.service
+
+import software.amazon.smithy.kotlin.codegen.core.KotlinDelegator
+import software.amazon.smithy.kotlin.codegen.core.KotlinDependency
+import software.amazon.smithy.kotlin.codegen.core.RuntimeTypes
+
+class ServiceStubGenerator(
+    private val packageName: String,
+    private val delegator: KotlinDelegator,
+) {
+
+    fun render() {
+        renderMainFile()
+        renderProtocolModule()
+        renderAuthModule()
+        renderConstraintValidators()
+        renderRouting()
+        renderPerOperationHandlers()
+    }
+
+    // Writes `Main.kt` that boots the embedded Ktor service.
+    private fun renderMainFile() {
+        delegator.useFileWriter("Main.kt", packageName) { writer ->
+
+            writer.addImport(RuntimeTypes.KtorServerCore.embeddedServer)
+            writer.addImport(RuntimeTypes.KtorServerNetty.Netty)
+            writer.dependencies.addAll(KotlinDependency.KTOR_LOGGING_BACKEND.dependencies)
+
+            writer.openBlock("public fun main(): Unit {")
+                .openBlock("embeddedServer(Netty, port = 8080) {")
```

Comment:
style: instead of adding the `embeddedServer` and `Netty` imports (L25-26) manually, you can reference it using a `#T` symbol, and the import gets added automatically

```
.openBlock("#T(#T, port = 8080) {", RuntimeTypes.KtorServerCore.embeddedServer, RuntimeTypes.KtorServerCore.Netty)
```